source/client-side-encryption/tests/legacy/fle2v2-CreateCollection.yml

```diff
@@ -1240,27 +890,53 @@ tests:
         - command_started_event:
             command:
               listCollections: 1
-              filter: { name: "encryptedCollection" }
+              filter: { name: *encrypted_collection_name }
             command_name: listCollections
             database_name: *database_name
         - command_started_event:
             command:
-              drop: "enxcol_.encryptedCollection.esc"
-            command_name: drop
-            database_name: *database_name
-        - command_started_event:
-            command:
-              drop: "enxcol_.encryptedCollection.ecc"
+              drop: *esc_collection_name
             command_name: drop
             database_name: *database_name
         - command_started_event:
             command:
-              drop: "enxcol_.encryptedCollection.ecoc"
+              drop: *ecoc_collection_name
             command_name: drop
             database_name: *database_name
         - command_started_event:
             command:
-              drop: "encryptedCollection"
+              drop: *encrypted_collection_name
             command_name: drop
             database_name: *database_name
         # events from dropCollection ... end
+  - description: "encryptedFields are consulted for metadata collection names"
+    clientOptions:
+      autoEncryptOpts:
+        kmsProviders:
+          aws: {} # Credentials filled in from environment.
+        encryptedFieldsMap:
+          default.encryptedCollection: {
```

Comment:
Same criticism as the above regarding JSON-like syntax. I'll leave this to you to address. The `fields` array can be copied over from an earlier suggestion for consistency.